pallets/detach/src/lib.rs

```diff
@@ -443,243 +510,256 @@ pub mod pallet {
+			for ed25519_key in &Crypto::ed25519_public_keys(KEY_TYPE) {
+				// Ed25519 key `ed25519_key` is not in `persisted_ed25119_keys`
+				if !persisted_ed25119_keys.contains(ed25519_key) {
+					let ecdsa_seed_hex =
+						Self::generate_ecdsa_seed_from_ed25519_key(&ed25519_key).unwrap();
+					log::debug!("Adding new key from seed: {:?}", ecdsa_seed_hex);
+					// `sp_io::crypto::ecdsa_generate` generates an ecdsa key for the given key type using an optional seed (in our case it's `Some(key_hex)`)
+					// and store it in the keystore. The seed needs to be a valid utf8.
+					// Source: https://paritytech.github.io/substrate/master/sp_io/crypto/fn.ecdsa_generate.html#
+					//
+					// Very Important Note: Since this function "store[s] it [i.e the eccdsa key] in the keystore",
+					// the ecdsa key will be retrievable when doing `sp_io::crypto::ecdsa_public_keys(KEY_TYPE)`
+					let _public =
+						Crypto::ecdsa_generate(KEY_TYPE, Some(ecdsa_seed_hex));
+					persisted_ed25119_keys.insert(*ed25519_key);
+					edited = true;
+				}
+			}
+			if edited {
+				// Update the storage value that is referred by `storage_ref` to `persisted_ed25119_keys`
+				storage_ref.set(&persisted_ed25119_keys);
+			}
+		}
+
+		/// Returns a Tuple of the following things:
+		/// 1. A **Signature** obtained by **signing the detach request `request` using a Fragnova-authorized account**.
 		///
-		/// Then, for each of the signed detach requests - send an unsigned transaction with a signed payload onto the Clamor Blockchain
-		/// (NOTE: the signed payload consists of a payload and a signature.
-		/// The payload is the detach request which is represented as a `DetachInternalData` struct (the struct contains the signature on the aforementioned tuple, amongst other things)
-		/// and the signature is the signature obtained from signing the aforementioned payload using `Signer::<T, T::AuthorityId>::any_account()`) (问Gio)
+		/// Note: Since it was signed by a Fragnova-authorized account, this signature can be presented to the External Blockchain `requests.target_chain`
+		/// to attach the "detached thing" (e.g a detached Proto-Fragment or a detached Fragment Instance) to the External Blockchain.
 		///
-		/// NOTE: `Signer::<T, T::AuthorityId>::any_account()` uses any of the keys that was added using the RPC `author_insertKey` into Clamor (https://polkadot.js.org/docs/substrate/rpc/#insertkeykeytype-text-suri-text-publickey-bytes-bytes)
-		fn process_detach_requests() {
+		/// 2. The Detach-Nonce of the detach request's target account. (See `DetachNonces` for more information).
+		fn get_detach_signature_and_detach_nonce(request: &DetachRequest) -> Result<(Vec<u8>, u64), Error<T>> {
+			match request.target_chain {
+				SupportedChains::EthereumMainnet
+				| SupportedChains::EthereumRinkeby
+				| SupportedChains::EthereumGoerli => {
+
+					Self::add_newly_found_ed25519_and_ecdsa_keys();
```

Comment:
@sinkingsugar why should we check for new Ed25519 keys for each iteration of `StorageValueRef::persistent(b"detach-requests")`. shouldn't we only check it once at the beginning of `detach.process_detach_requests()`?

Here's how it is done in the current code for reference (my question might make more sense when you see this 😅):
https://github.com/fragcolor-xyz/clamor/blob/devel/pallets/detach/src/lib.rs#L485-L514

Reply:
Here's the updated link: https://github.com/fragcolor-xyz/clamor/pull/215/files#diff-f3de07c470ea31d8f5f62b2949c157b04d3506ecbf23fe6daf37898f05769499L485-L514